docs/stdlib/json.rst

```diff
@@ -353,6 +356,105 @@ possible to cast a JSON value directly into a :eql:type:`tuple`.
 ----------
 
 
+.. eql:function:: std::json_set( \
+                    target: json, \
+                    variadic path: str, \
+                    named only value: optional json, \
+                    named only create_if_missing: bool = true, \
+                    named only empty_treatment: JsonSetEmptyTreatment = \
+                      JsonSetEmptyTreatment.ReturnEmpty) \
+                  -> optional json
+
+    Return an updated JSON target with a new value.
+
+    .. warning::
+
+      This function is only available in EdgeDB 2.0 or later.
+
+    .. code-block:: edgeql-repl
+
+        db> select json_set(
+        ...   to_json('{"a": 10, "b": 20}'),
+        ...   'a',
+        ...   value := <json>true,
+        ... );
+        {'{"a": true, "b": 20}'}
+        db> select json_set(
+        ...   to_json('{"a": {"b": {}}}'),
+        ...   'a', 'b', 'c',
+        ...   value := <json>42,
+        ... );
+        {'{"a": {"b": {"c": 42}}}'}
+
+    If ``create_if_missing`` is set to ``false``, a new path for the value
+    won't be created.
+
+    .. code-block:: edgeql-repl
+
+        db> select json_set(
+        ...   to_json('{"a": 10, "b": 20}'),
+        ...   'с',
+        ...   value := <json>42,
+        ... );
+        {'{"a": 10, "b": 20, "c": 42}'}
+        db> select json_set(
+        ...   to_json('{"a": 10, "b": 20}'),
+        ...   'с',
+        ...   value := <json>42,
+        ...   create_if_missing := false,
+        ... );
+        {'{"a": 10, "b": 20}'}
+
+    ``empty_treatment`` is an enumeration responsible for the behavior of the
+    function if an empty set is passed to ``new_value``. It contains one of
+    the following values:
+
+    - ``ReturnEmpty``: return empty set, default
+    - ``ReturnTarget``: return ``target`` unmodified
+    - ``RaiseException``: raise an ``InvalidValueError``
+    - ``UseJsonNull``: use a ``null`` JSON value
+    - ``DeleteKey``: delete the object key
+
+    .. code-block:: edgeql-repl
+
+        db> select json_set(
+        ...   to_json('{"a": 10, "b": 20}'),
+        ...   'a',
+        ...   value := <json>{}
+        ... );
+        {}
+        db> select json_set(
+        ...   to_json('{"a": 10, "b": 20}'),
+        ...   'a',
+        ...   value := <json>{},
+        ...   empty_treatment := JsonSetEmptyTreatment.ReturnTarget,
+        ... );
+        {'{"a": 10, "b": 20}'}
+        db> select json_set(
+        ...   to_json('{"a": 10, "b": 20}'),
+        ...   'a',
+        ...   value := <json>{},
+        ...   empty_treatment := JsonSetEmptyTreatment.RaiseException,
```

Comment:
@elprans How about we rename `JsonSetEmptyTreatment` to `JsonEmpty`? The former is offensively long to type.

```suggestion
        ...   empty_treatment := JsonEmpty.RaiseException,
```

Reply:
On that note, `JsonEmpty.Error` is better than `JsonEmpty.RaiseException`

Reply:
I'm +1 on these changes, I think